scripts/emulator-testing/emulators/firestore-emulator.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Emulator } from './emulator';
+
+export class FirestoreEmulator extends Emulator {
+  projectId: string;
+
+  constructor(port = 8087, projectId = 'test-emulator') {
+    super(port);
+    this.projectId = projectId;
+    this.binaryName = 'firestore-emulator.jar';
+    this.binaryUrl =
+      'https://storage.googleapis.com/firebase-preview-drop/emulator/cloud-firestore-emulator-v1.2.1.jar';
```

Comment:
I'm a little concerned about the hard coded url. If we forgot to update the version, we will eventually become using an out dated emulator binary. @ryanpbrewster Is there any way to always get the latest binary?

Reply:
There was some discussion before on specifying emulator versions.

The conclusion seemed to be that it's probably better to have a locked version and update the version manually, than to always get the latest version and have risk that test may fail surprisingly with the new version.

Reply:
Okay. That sounds fine to me. Can you please add a comment about it and include a pointer to where the latest emulator can be found?

Reply:
Done.